### DISCUSSION TITLE
Detecting when a websocket connection is down

### DISCUSSION THREAD

**Discussion by kpysniak** [enisdenjo/graphql-ws#493] [Q&A] [answered]
I am currently connecting to my GraphQL server using apollo client and graphql-ws:
```javascript
    const wsLink = new GraphQLWsLink(
      createClient({
        url: process.env.NEXT_PUBLIC_WEBSOCKET_DATAGATE_ADDRESS,
        reconnect: true,
        lazy: false,
        connectionParams: () => {
          return {
            headers: {
              Authorization: token,
            },
          };
        },
      })
    );

    this.apolloClient = new ApolloClient({
      link: wsLink,
      cache: new InMemoryCache(),
    });
``` 

Is there a way to detect when the connection of GraphQLWsLink gets broken and force reconnect?

**Comment by enisdenjo** [comment, marked as answer]
The client automatically performs silent reconnects when the connection is abruptly closed. By default, it [retries 5 times](https://the-guild.dev/graphql/ws/docs/interfaces/client.ClientOptions#retryattempts) using [randomised exponential backoff](https://the-guild.dev/graphql/ws/docs/interfaces/client.ClientOptions#retrywait). You can disable the retries by setting the `retryAttempts` client option to zero (0).

Furthermore, you can [listen for the closed event](https://the-guild.dev/graphql/ws/docs/modules/client#eventclosedlistener) for when the WebSocket connection closes. 

P.S. there is no `reconnect` option in the client.

```diff
    const wsLink = new GraphQLWsLink(
      createClient({
        url: process.env.NEXT_PUBLIC_WEBSOCKET_DATAGATE_ADDRESS,
-       reconnect: true,
        lazy: false,
        connectionParams: () => {
          return {
            headers: {
              Authorization: token,
            },
          };
        },
      })
    );
```
